src/tests/ftest/recovery/pass_0.py

```diff
@@ -0,0 +1,155 @@
+"""
+  (C) Copyright 2023 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+from apricot import TestWithServers
+from general_utils import report_errors
+from exception_utils import CommandFailure
+
+
+class Pass0Test(TestWithServers):
+    """Test Pass 0: Management Service & Membership
+
+    :avocado: recursive
+    """
+
+    def test_checker_on_admin_excluded(self):
+        """Test checker can only be run when the system status is AdminExcluded.
+
+        1. Call dmg check enable.
+        2. Stop rank 1.
+        3. Call dmg system query and check that status of at least one rank is not
+        "checkerstarted". We verify that the new checkerstarted state is properly changed.
+        4. Call dmg check start. It should show error because the stopped rank is not at
+        CheckerStarted state.
+        5. Call dmg check query. It should show error because the stopped rank is not at
+        CheckerStarted state.
+        6. Call dmg check disable. It should work.
+        7. Restart the stopped rank for cleanup.
+
+        Jira ID: DAOS-11703
+
+        :avocado: tags=all,weekly_regression
+        :avocado: tags=vm
+        :avocado: tags=recovery,pass_0
+        :avocado: tags=test_checker_on_admin_excluded
+        """
+        errors = []
+        dmg_command = self.get_dmg_command()
+
+        # 1. Call dmg check enable.
+        dmg_command.check_enable()
+
+        # 2. Stop rank 1.
+        dmg_command.system_stop(ranks="1")
+
+        # 3. Call dmg system query and check that status of at least one rank is not
+        # "checkerstarted". We verify that the new checkerstarted state is properly
+        # changed.
+        query_out = dmg_command.system_query()
+        not_checkerstarted_found = False
+        for member in query_out["response"]["members"]:
+            if member["state"] != "checkerstarted":
+                not_checkerstarted_found = True
+                break
+        if not not_checkerstarted_found:
+            # All rank's status is "checkerstarted".
+            errors.append("All rank's status is checkerstarted!")
```

Comment:
Would it be better to fail immediately?

Reply:
Yes. If all rank's status is `checkerstarted`, it'll affect the next steps, so it's better to fail here immediately. Fixed.